src/paginator.js

```diff
@@ -0,0 +1,300 @@
+import assign from 'lodash.assign';
+import omit from 'lodash.omit';
+
+
+export const styles = Object.freeze({
+  pageNumber: 1,
+  limitOffset: 2,
+});
+
+export const errors = Object.freeze({
+  invalidStyle: 'Invalid pagination style',
+  calculateOffsetNoLimit: 'Can\'t calculate offsets without a limit',
+  calculatePageNoLimit: 'Can\'t calculate pages without a limit'
+});
+
+const getResponseBody = function(response) {
+  return response.json();
+};
+
+const defaultOptions = {
+  limitQueryParam: 'limit',
+  offsetQueryParam: 'offset',
+  pageQueryParam: 'page',
+  style: styles.pageNumber
+};
+
+const defaultQueryParams = {
+  limit: null,
+  offset: 0,
+  page: 1
+};
+
+export class Paginator {
+  constructor(request, requestOptions = null, queryParams = {}, options = {}) {
```

Comment:
We wanted to make it match up with a request's parameters. A request takes two, `options` and `queryParams`, so Paginator should take at least three.

I realize this has four, but see my other comment I'm about to make.